crates/anvil/src/eth/api.rs

```diff
@@ -941,32 +919,29 @@ impl EthApi {
             // but EIP-1559 prepends a version byte, so we need to encode the data first to get a
             // valid rlp and then rlp decode impl of `TypedTransaction` will remove and check the
             // version byte
-            let extend = rlp::encode(&data);
-            let tx = match rlp::decode::<TypedTransaction>(&extend[..]) {
+            let extend = alloy_rlp::encode(data);
+            let tx = match TypedTransaction::decode(&mut &extend[..]) {
```

Comment:
do we have enough tests for typed transaction decoding?
can't find any eip1559 specific tests in eth/transaction/mod

Reply:
We do have eip1559 tests: https://github.com/foundry-rs/foundry/blob/evalir/migrate-base-anvil/crates/anvil/core/src/eth/transaction/mod.rs#L1257-L1275